src/zenml/integrations/hyperai/service_connectors/hyperai_service_connector.py

```diff
@@ -0,0 +1,386 @@
+
+        Args:
+            kwargs: Additional implementation specific keyword arguments to pass
+                to the session or client constructor.
+
+        Returns:
+            An authenticated Paramiko SSH client.
+        """
+        logger.info("Connecting to HyperAI instance...")
+        assert self.resource_id is not None
+
+        try:
+            paramiko_client = self._create_paramiko_client()
+            if paramiko_client is None:
+                raise RuntimeError(
+                    "Could not connect to instance due to missing paramiko client."
+                )
+            else:
+                return paramiko_client
+        except Exception as e:
+            logger.error(
+                "Unknown error while connecting to HyperAI instance: %s. Please check your network connection, IP address, and authentication details.",
+                e,
+            )
+
+    def _configure_local_client(
+        self,
+        **kwargs: Any,
+    ) -> None:
+        """There is no local client for the HyperAI connector, so it does nothing."""
+        pass
+
+    @classmethod
+    def _auto_configure(
+        cls,
+        auth_method: Optional[str] = None,
+        resource_type: Optional[str] = None,
+        resource_id: Optional[str] = None,
+        **kwargs: Any,
+    ) -> "HyperAIServiceConnector":
+        """Auto-configure the connector.
+
+        Not supported by the HyperAI connector.
+
+        Args:
+            auth_method: The particular authentication method to use. If not
+                specified, the connector implementation must decide which
+                authentication method to use or raise an exception.
+            resource_type: The type of resource to configure.
+            resource_id: The ID of the resource to configure. The
+                implementation may choose to either require or ignore this
+                parameter if it does not support or detect an resource type that
+                supports multiple instances.
+            kwargs: Additional implementation specific keyword arguments to use.
+
+        Raises:
+            NotImplementedError: If the connector auto-configuration fails or
+                is not supported.
+        """
+        raise NotImplementedError(
+            "Auto-configuration is not supported by the HyperAI connector."
+        )
+
+    def _verify(
+        self,
+        resource_type: Optional[str] = None,
+        resource_id: Optional[str] = None,
+    ) -> List[str]:
+        """Verify that a connection can be established to the HyperAI instance.
+
+        Args:
+            resource_type: The type of resource to verify. Must be set to the
+                Docker resource type.
+            resource_id: The HyperAI instance to verify.
+
+        Returns:
+            The resource ID if the connection can be established.
+        """
+        self._authorize_client()
```

Comment:
same here: please raise `AuthorizationException` if any errors occur related to misconfigured or invalid credentials. Or just let `AuthorizationException` fall through from `self._authorize_client()`